clang/lib/Headers/hlsl/hlsl_intrinsics.h

```diff
@@ -175,6 +175,34 @@ const inline float distance(__detail::HLSL_FIXED_VECTOR<float, N> X,
   return __detail::distance_vec_impl(X, Y);
 }
 
+//===----------------------------------------------------------------------===//
+// dst builtins
+//===----------------------------------------------------------------------===//
+
+/// \fn fvector dst(fvector, fvector)
+/// \brief Calculates a distance vector.
+/// \param Src0 [in] The first vector contains the squared distance represented
+/// as {_, d*d, d*d, _}
+/// \param Src1 [in] The second vector contains the reciprocal distance
+/// represented as  {_, 1/d, _, 1/d}
+///
+/// Return the computed distance vector contain {1, d, d*d, 1/d}
+
+_HLSL_16BIT_AVAILABILITY(shadermodel, 6.2)
+const inline vector<half, 4> dst(vector<half, 4> Src0, vector<half, 4> Src1) {
```

Comment:
We tend to use the more common typedefs unless we're using a template-dependent vector.
```suggestion
const inline half4 dst(half4 Src0, half4 Src1) {
```